spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -272,9 +274,40 @@ private Entity convertToEntityForSave(Object entity, Key... ancestors) {
 		Builder builder = Entity.newBuilder(key);
 		this.datastoreEntityConverter.write(entity, builder);
 		saveDescendents(entity, key);
+		saveReferences(entity, builder);
 		return builder.build();
 	}
 
+	private void saveReferences(Object entity, Builder builder) {
+		DatastorePersistentEntity datastorePersistentEntity = this.datastoreMappingContext
+				.getPersistentEntity(entity.getClass());
+		datastorePersistentEntity.doWithReferenceProperties(persistentProperty -> {
+			PersistentPropertyAccessor accessor = datastorePersistentEntity.getPropertyAccessor(entity);
+			Object val = accessor.getProperty(persistentProperty);
+			if (val == null) {
+				return;
+			}
+			Value<?> value;
+			if (persistentProperty.isCollectionLike()) {
+				if (val.getClass().isArray() && val.getClass() != byte[].class) {
```

Comment:
We're using this check so often  I think we should make this into a method for `DatastorePersistentProperty` called `isNonBytesArray`